src/AutoRest.CSharp/Common/Input/Utf8JsonReaderExtensions.cs

```diff
@@ -70,6 +70,46 @@ public static bool TryReadString(this ref Utf8JsonReader reader, string property
             return true;
         }
 
+        public static bool TryReadEnumValue(this ref Utf8JsonReader reader, string propertyName, ref object? value)
+        {
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException();
+            }
+
+            if (reader.GetString() != propertyName)
+            {
+                return false;
+            }
+
+            reader.Read();
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.String:
+                    value = reader.GetString() ?? throw new JsonException();
```

Comment:
Better to provide some information in the exception to help trouble shooting.

Reply:
done